File: src/main.java.snake/snake_03/under_03/test04_05py/Item05.py
class Item:

    def __init__( self ):
        self._id = None
        self._name = None
        self._price = None
        self._tax = None
    
    @property
    def id( self ):
        return self._id

    @id.setter
    def id( self, value ):
        self._id = value
        return self     # 自分自身を返す
    
    @property
    def name( self ):
        return self._name

    @name.setter
    def name( self, value ):
        self._name = value
        return self     # 自分自身を返す
    
    @property
    def price( self ):
        return  self._price

    @price.setter
    def price( self, value ):
        self._price = value
        return self     # 自分自身を返す
    
    @property
    def tax( self ):
        return self._tax

    @tax.setter
    def tax( self, value ):
        self._tax = value
        return self     # 自分自身を返す

File: src/main.java.snake/snake_03/under_03/test04_05py/Item05.java
package snake_01.src.main.java.snake.snake_03.under_03.test04_05;

/* private を付けないメンバー変数は コンストラクタで作成できるprivateを付けた場合はgetter/setter メソッドが必要 外部からアクセス出来なくなる為
*/
public class Item05
{
  private String id;
  private String name;
  private double price;
  private double tax;

/*コンストラクタの場合はこれでOK
  public Item05( String id, String name, double price, double tax ){
    this.id = id;
    this.name = name;
    this.price = price;
    this.tax = tax;

  }
*/

  public void setId( String id ){
    this.id = id;/*private を付けた場合は上からコピペ*/
  }

  public void setName( String name ){
    this.name = name;
  }

  public void setPrice( double price ){
    this.price = price;
  }

  public void setTax( double tax ){
    this.tax = tax;
  }

  public static void save( ){
    indi( "save :" );
  }




  public String toString(){
    return( "Item05 :" + "id=" + id + " " + "name=" + name + " " + "price=" + price + " " + "tax=" + tax );
  }

  public static void indi( String s0 ){
    System.out.println( s0 );
  }
}
